### DISCUSSION TITLE
Session 5- Criando um site estático - aula 26- Buckets

### DISCUSSION THREAD

**Discussion by edwardborges** [chgasparoto/curso-aws-com-terraform#126] [Q&A] [answered]
Professor Bom dia
Pode ajudar? Eu estava criando conforme a aula os recurso,mas ao fazer o **$ terraform plan**:

Ele me informa que vai criar 19 recursos e não 4, como acontece no seu vídeo minuto 10:01

Segue em anexo meu plan.out  o meu terraform plan. 

[plan-site estático.txt](https://github.com/chgasparoto/curso-aws-com-terraform/files/10154443/plan-site.estatico.txt)

Como pode ver ele cria vários modulos do wesite( esse aquivo eu montei de acordo com seu repositório ).

Segue a baixo o s3.tf do meu terraform 

``` terraform
data "template_file" "s3-public-policy" {
  template = file("policy.json")
  vars = {
    bucket_name = local.domain

  }
}

module "logs" {
  source        = "github.com/chgasparoto/terraform-s3-object-notification"
  name          = "${local.domain}-logs"
  acl           = "log-delivery-write"
  force_destroy = !local.has_domain
  tags          = local.common_tags
}

module "website" {
  source        = "github.com/chgasparoto/terraform-s3-object-notification"
  name          = local.domain
  acl           = "public-read"
  policy        = data.template_file.s3-public-policy.rendered
  force_destroy = !local.has_domain
  tags          = local.common_tags

  versioning = {
    enabled = true
  }

  filepath = "${local.website_filepath}/build"
  website = {
    index_document = "index.html"
    error_document = "index.html"
  }

  logging = {
    target_bucket = module.logs.name
    target_prefix = "access/"
  }
}

module "redirect" {
  source        = "github.com/chgasparoto/terraform-s3-object-notification"
  name          = "www.${local.domain}"
  acl           = "public-read"
  force_destroy = !local.has_domain
  tags          = local.common_tags

  website = {
    redirect_all_requests_to = local.has_domain ? var.domain : module.website.website
  }
}
```

 O que eu errei? pode ajudar?

**Comment by chgasparoto** [comment, marked as answer]
Fala @edwardborges , o seu código e output está correto. No tempo que você informou que eu rodei o comando no vídeo eu não tinha gerado os arquivos com o comando `npm run build`. Por se tratar de versões diferentes, é esperado que o número de arquivos não seja exatamente o mesmo.

**Comment by edwardborges** [comment]
Bom dia Professor
Isso mesmo
Porém agora o problema é no terraform apply
ele informa que há um erro no modulo de logs

´´´´´´
│ Error: creating Amazon S3 (Simple Storage) Bucket (default-logs): BucketAlreadyExists: The requested bucket name is not available. The bucket namespace is shared by all users of the system. Please select a different name and try again.
│       status code: 409, request id: DDX86D6G1N8QY3BF, host id: ESEQrPNihKH52wYcTjqmNZkPrRmAOujGTIO7hg5ZI32FI18J/QbN+1ZBwbunYCq/R1DCHqy3erE=
│ 
│   with module.logs.aws_s3_bucket.this,
│   on .terraform/modules/logs/main.tf line 1, in resource "aws_s3_bucket" "this":
│    1: resource "aws_s3_bucket" "this" {
´´´´

Na mensagem acima estáo plan que te enviei 
Mas esse plan e nem o validate informa um erro.

Onde estou errando na configuração?

Desde já agradeço retorno

**Reply by edwardborges** [reply]
Professor Bom dia
O que entendi foi que não consigo criar um bucket por conta do nome parace já existir
mas estou puxando é o modulo do logs que você informa na aula
ela não tinha que ser uma variavél randompet?

Quando olho o arquivo de variables.tf ( pasta logs)

Mostra normal 


```terraform
variable "name" {
  description = "Bucket unique name. It can contain only number, letter and dash."
  type        = string
}
```

Agora fiquei confuso

**Reply by chgasparoto** [reply]
Fala @edwardborges , você está usando um nome bem comum para o seus buckets, veja a mensagem de erro que você postou acima.

> Error: creating Amazon S3 (Simple Storage) Bucket (default-logs): BucketAlreadyExists

De alguma forma você está passando como prefixo a palavra `default`. Verifique seus locals para ver como este `local.domain` está sendo setado.

**Reply by edwardborges** [reply]
Bom dia Professor

Agradeço o retorno
Isso que estava matutando , se o locals estava setando pra o main , mas não identifiquei

Segue meu código abaixo:
```terraform
locals {
  has_domain       = var.domain != ""
  domain           = local.has_domain ? var.domain : random_pet.website.id
  regional_domain  = module.website.regional_domain_name
  website_filepath = "${path.module}/../website"

  common_tags = {
    Project   = "Curso AWS com Terraform"
    Service   = "Static Website"
    CreatedAt = "2022-12-01"
    Module    = "3"
  }
}
```

**Comment by edwardborges** [comment]
Professor acho que achei o ERRO 

No arquivo de variable.tf da minha pasta local 
``` terraform
variable "domain" {
  type        = string
  description = ""
  default     = ""
}
```

Tava escrito default ai revendo o vídeo, você deixa sem descrição
dei o terraform apply e deu certo.
